server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -934,17 +934,33 @@ static Settings aggregateIndexSettings(
      * @param settingsBuilder index settings builder to be updated with relevant settings
      * @param requestSettings settings passed in during index create request
      * @param clusterSettings cluster level settings
+     * @param combinedTemplateSettings combined template settings which satisfy the index
      */
-    private static void updateReplicationStrategy(Settings.Builder settingsBuilder, Settings requestSettings, Settings clusterSettings) {
+    private static void updateReplicationStrategy(
+        Settings.Builder settingsBuilder,
+        Settings requestSettings,
+        Settings clusterSettings,
+        Settings combinedTemplateSettings
+    ) {
+        // The replication setting is applied in the following order:
+        // i. Explicit index creation request parameter
+        // ii. Defaults to segment if remote store attributes on the cluster
+        // iii. Template property for replication type
+        // iv. Default cluster level setting
+
+        final ReplicationType indexReplicationType;
         if (INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings)) {
-            settingsBuilder.put(SETTING_REPLICATION_TYPE, INDEX_REPLICATION_TYPE_SETTING.get(requestSettings));
+            indexReplicationType = INDEX_REPLICATION_TYPE_SETTING.get(requestSettings);
+        } else if (INDEX_REPLICATION_TYPE_SETTING.exists(combinedTemplateSettings)) {
+            indexReplicationType = INDEX_REPLICATION_TYPE_SETTING.get(combinedTemplateSettings);
         } else if (CLUSTER_REPLICATION_TYPE_SETTING.exists(clusterSettings)) {
-            settingsBuilder.put(SETTING_REPLICATION_TYPE, CLUSTER_REPLICATION_TYPE_SETTING.get(clusterSettings));
+            indexReplicationType = CLUSTER_REPLICATION_TYPE_SETTING.get(clusterSettings);
         } else if (isRemoteStoreAttributePresent(clusterSettings)) {
```

Comment:
I think `updateRemoteStoreSettings` will check this and overwrite whatever this method sets correct? If so maybe we can remove the redundant check here or better yet do the remote store check up front and skip this entirely?

Reply:
hmm I'm wrong here
```
    private static void updateRemoteStoreSettings(Settings.Builder settingsBuilder, Settings clusterSettings) {
        if (isRemoteStoreAttributePresent(clusterSettings)) {
            settingsBuilder.put(SETTING_REMOTE_STORE_ENABLED, true)
                .put(
                    SETTING_REMOTE_SEGMENT_STORE_REPOSITORY,
                    clusterSettings.get(
                        Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY
                    )
                )
                .put(
                    SETTING_REMOTE_TRANSLOG_STORE_REPOSITORY,
                    clusterSettings.get(
                        Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY
                    )
                );
        }
    }
```